assert/assertions_test.go

```diff
@@ -2816,6 +2816,42 @@ func TestEventuallyIssue805(t *testing.T) {
 	})
 }
 
+func TestEventuallySucceedQuickly(t *testing.T) {
+	mockT := new(testing.T)
+
+	condition := func() bool { <-time.After(time.Millisecond); return true }
+
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		True(t, Eventually(mockT, condition, 1000*time.Millisecond, 100*time.Millisecond))
+	}()
+
+	select {
+	case <-done:
+	case <-time.After(10 * time.Millisecond):
+		Fail(t, `condition not satisfied quickly enough`)
+	}
+}
+
+func TestEventuallyWithTSucceedQuickly(t *testing.T) {
+	mockT := new(testing.T)
+
+	condition := func(t *CollectT) { <-time.After(time.Millisecond) }
+
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		True(t, EventuallyWithT(mockT, condition, 1000*time.Millisecond, 100*time.Millisecond))
+	}()
+
+	select {
+	case <-done:
+	case <-time.After(10 * time.Millisecond):
+		Fail(t, `condition not satisfied quickly enough`)
+	}
```

Comment:
I think you could simplify the test by just checking with `tick` longer than `waitFor`:
```suggestion
	condition := func(t *CollectT) {}

	True(t, EventuallyWithT(mockT, condition, 100*time.Millisecond, time.Second))
```

Reply:
Great idea!